### DISCUSSION TITLE
Automatic empty ExFAT partition

### DISCUSSION THREAD

**Discussion by joseph85750** [finnix/finnix#46] [Ideas]
Having the USB automatically create an empty ExFAT partition with the remaining space not needed for the distro itself.  I've done this manually with other bootable USB distros, so it would be nice if it would happen as part of the build.  Basically, if I have a 16G USB stick, it can go wasted due to the entire distro being on a FAT32 partition.  My idea is to make the image such that it installs the base OS on a smaller FAT32 partition (ie: 1G), and format the rest of the USB stick as ExFAT.   I pick ExFAT because it's readable and writable for Linux, Mac, and Windows.  Plus it doesn't suffer from the 2G file size restriction.
Since the initial DD process would have no way of knowing, it would need to be done after the initial boot.  It would boot once, determine what the size of the USB stick is, and create the next partition with the remaining space.  With this, not only would the USB be bootable, but other rescue images/files could be stored on the USB stick.  I've done this sort of thing to make bootable USB installers, where I store a gzipped disk image on my extra partition, and can DD it to the host system after booting this USB distro.

**Comment by pstray** [comment]
Maybe a smaller overlayed partition for keeping small changes to the finnix setup, like persistent storage of keyboard layout, shell history, and small scripts would be useful.

**Comment by joseph85750** [comment]
Actually I found a distro which does this:   [Puppy Lunux](https://puppylinux-woof-ce.github.io/)
After you DD to your USB, and boot, it only uses about 5G.  It leaves the rest unformatted.  So you can fdisk the remaining space to be whatever you want (ext4, extfat, etc).
